src/Workspaces/Core/Portable/Workspace/Solution/SolutionState.CompilationTracker.cs

```diff
@@ -521,14 +499,6 @@ private async Task<Compilation> GetOrBuildDeclarationCompilationAsync(SolutionSe
 
                 if (compilation == null)
                 {
-                    // this can happen if compilation is already kicked out from the cache.
-                    // check whether the state we have support declaration only compilation
-                    if (state.DeclarationOnlyCompilation != null)
-                    {
-                        // we have declaration only compilation. build final one from it.
-                        return FinalizeCompilationAsync(solution, state.DeclarationOnlyCompilation, authoritativeGeneratedDocuments, nonAuthoritativeGeneratedDocuments, compilationWithStaleGeneratedTrees: null, generatorDriver, cancellationToken);
-                    }
-
                     // We've got nothing.  Build it from scratch :(
                     return BuildCompilationInfoFromScratchAsync(solution, cancellationToken);
```

Comment:
Ditto to earlier comment, are we now going to re-run generators since we're not passing the previously ran results?

Reply:
so let's chat on monday abou tthis.  I think we should instead have an explicit way to pass source generated docs along that is very explicit in what we're doing and what the lifetimes are of these things.  I don't think (to me at least) that it was even intentional or known that it would be the 'decl only compilation' that would be keeping those docs around as a side effect.